src/systems/systems/src/shared/mod.rs

```diff
@@ -576,99 +577,91 @@ mod delayed_pulse_true_logic_gate_tests {
             self.expression_result = value;
         }
 
-        fn gate_output(&mut self) -> bool {
+        fn gate_output(&self) -> bool {
             self.gate.output()
         }
     }
     impl Aircraft for TestAircraft {
-        fn update_before_power_distribution(&mut self, context: &UpdateContext) {
+        fn update_before_power_distribution(
+            &mut self,
+            context: &UpdateContext,
+            _: &mut Electricity,
+        ) {
             self.gate.update(context, self.expression_result);
         }
     }
     impl SimulationElement for TestAircraft {}
 
     #[test]
     fn when_the_expression_is_false_initially_returns_false() {
-        let mut aircraft =
-            TestAircraft::new(DelayedPulseTrueLogicGate::new(Duration::from_millis(100)));
-        let mut test_bed = SimulationTestBed::new_with_delta(Duration::from_millis(1_000));
+        let mut test_bed = SimulationTestBed::new(|_| {
+            TestAircraft::new(DelayedPulseTrueLogicGate::new(Duration::from_millis(100)))
+        });
 
-        test_bed.run_aircraft(&mut aircraft);
+        test_bed.run_with_delta(Duration::from_millis(1_000));
```

Comment:
`run` runs a 1 second tick.
```suggestion
        test_bed.run();
```